### ISSUE TITLE
Install Vale in GitHub Action

### ISSUE THREAD

**Issue by stevepiercy** [plone/documentation#1392]
To use the GitHub Action, I would need to translate the command from Makefile to get the list of files to lint, and set an environment variable:

`VALEFILES       := $(shell find -L $(DOCS_DIR) -type d \( -path $(DOCS_DIR)/plone.restapi/lib/* -o  -path $(DOCS_DIR)"/plone.restapi/performance/*" \) -prune -false -o -type f -name "*.md" -print)`

How would I translate that to GitHub Actions? I have no idea.

It might be easier to just install Vale where all the other tests are run in CI and use `make vale`. [Narrator: it wasn't](https://github.com/plone/documentation/pull/1391).

There is also an option to use a [GitHub Action of Vale](https://github.com/errata-ai/vale-action), and I might try that as well.

_Originally posted by @stevepiercy in https://github.com/plone/documentation/issues/1366#issuecomment-1357549265_

**Comment by davisagli** [comment]
@stevepiercy I think it should be a GHA step that looks like this:
```
- run: echo "VALEFILES=$(find -L ./docs/ -type d \( -path ./docs/plone.restapi/lib/* -o -path "./docs/plone.restapi/performance/*" \) -prune -false -o -type f -name "*.md" -print)" >> $GITHUB_ENV
```

but you might need to add another backslash before each of the 2 existing backslashes

**Comment by stevepiercy** [comment]
@davisagli I gave it a shot in https://github.com/plone/documentation/pull/1391, based on https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

It fails: https://github.com/plone/documentation/actions/runs/3738520751/jobs/6344687632

However, were you talking about modifying the open PR or trying to run Vale in a Docker image?

**Comment by davisagli** [comment]
@stevepiercy I commented there with another suggestion.

I definitely don't have the full context here, and was just responding to this issue, which asks about github actions. It doesn't say anything about Docker...

**Comment by stevepiercy** [comment]
Yeah, here I am considering two ways to run Vale in GHA: via Docker image or install Vale into the test environment. I tried the latter, fumbled around and your suggestion _almost_ worked, after I removed the Microsoft styles.

The `find` command returns only 86 `.md` files because it did not traverse the symlinks via `-L`. We need to find 246. I'll fiddle around with the syntax some more.

**Comment by stevepiercy** [comment]
Here's a partial desired result for those 86 .md files:
https://github.com/plone/documentation/actions/runs/3738587320/jobs/6344820533#step:5:1281

It errors out as expected, returning misspelled words.

**Comment by stevepiercy** [comment]
`make deps` installs the docs and makes the symlinks. That's why the submodule files were not found. All good there now.

Now let's try to get those Microsoft styles installed...

**Comment by stevepiercy** [comment]
I learned that GitHub Actions does not support `cp -r`. That is awful. I resorted to this SO tip:

https://stackoverflow.com/questions/62432172/github-action-to-copy-and-a-rename-a-file

**Comment by davisagli** [comment]
@stevepiercy You used an en dash instead of a hyphen for `-r`. I only noticed because I was super weirded out and copied the error message to search for what was going on, and found someone else who had done the same thing.

**Comment by stevepiercy** [comment]
Argh, fat fingers! That must have happened when I tried `-R` first, holding down the shift key. Fixed in [3daf0a6](https://github.com/plone/documentation/pull/1391/commits/3daf0a6a451fd81be04d6054a866e31577fd8802). Thank you @davisagli

**Comment by stevepiercy** [comment]
Closed by https://github.com/plone/documentation/pull/1391
